runtime/j9vm/javanextvmi.cpp

```diff
@@ -253,29 +253,24 @@ JVM_VirtualThreadMountBegin(JNIEnv *env, jobject thread, jboolean firstMount)
 	J9VMThread *currentThread = (J9VMThread *)env;
 	J9JavaVM *vm = currentThread->javaVM;
 	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	BOOLEAN releasedVirtualThreadInspectorMutex = FALSE;
 
 	vmFuncs->internalEnterVMFromJNI(currentThread);
-	f_monitorEnter(vm->liveVirtualThreadListMutex);
 	j9object_t threadObj = J9_JNI_UNWRAP_REFERENCE(thread);
 
 	assert(IS_VIRTUAL_THREAD(currentThread, threadObj));
 
-	while (0 != J9OBJECT_I64_LOAD(currentThread, threadObj, vm->virtualThreadInspectorCountOffset)) {
-		/* Thread is being inspected or unmounted, wait. */
-		VM_VMHelpers::pushObjectInSpecialFrame(currentThread, threadObj);
-		vmFuncs->internalExitVMToJNI(currentThread);
-		f_monitorWait(vm->liveVirtualThreadListMutex);
-		vmFuncs->internalEnterVMFromJNI(currentThread);
-		threadObj = VM_VMHelpers::popObjectInSpecialFrame(currentThread);
-	}
-
-	/* Prevent inspectors from inspecting this thread during stack swap and mount by locking from notifyJvmtiMountBegin
-	 * to notifyJvmtiMountEnd. See getVMThread() in jvmtiHelpers.c.
-	 */
-	J9OBJECT_I64_STORE(currentThread, threadObj, vm->virtualThreadInspectorCountOffset, -1);
-
 	if (firstMount) {
-		if (NULL == vm->liveVirtualThreadList) {
+		f_monitorEnter(vm->virtualThreadListMutex);
+		while (vm->inspectVirtualThreadList) {
+			/* Virtual thread list is being inspected, wait. */
+			VM_VMHelpers::pushObjectInSpecialFrame(currentThread, threadObj);
+			vmFuncs->internalExitVMToJNI(currentThread);
+			f_monitorWait(vm->virtualThreadListMutex);
+			vmFuncs->internalEnterVMFromJNI(currentThread);
+			threadObj = VM_VMHelpers::popObjectInSpecialFrame(currentThread);
```

Comment:
If you want to save a little code, you could avoid the push/pop and just unwrap the JNI ref again here.